### DISCUSSION TITLE
Receive 404 Not Found errors during zip deploy every so often

### DISCUSSION THREAD

**Discussion by kamranayub** [Azure/functions-action#208] [Q&A]
I've been working most of the day with an Azure Function and have done about 20+ deployments. It seems like every other one, or once every 3-4 times, I get the following error. If I just re-run the failed job, it'll work.

Node v18 / Linux Consumption Plan / Oryx Build / SCM Do Build During Deployment

Has anyone experienced this?

```
Run Azure/functions-action@v1
  with:
    app-name: app-name
    package: ./src/functions
    publish-profile: ***
    respect-funcignore: true
    enable-oryx-build: true
    scm-do-build-during-deployment: true
    respect-pom-xml: false
  env:
    AZURE_FUNCTIONAPP_PACKAGE_PATH: ./src/functions

    NPM_CONFIG_USERCONFIG: /home/runner/work/_temp/.npmrc

Successfully parsed SCM credential from publish-profile format.
Using SCM credential for authentication, GitHub Action will not perform resource validation.
(node:1917) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
Successfully acquired app settings from function app (with SCM credential)!
Will archive ./src/functions into /home/runner/work/_temp/temp_web_package_00564).zip as function app content
Will use Kudu https://<scmsite>/api/zipdeploy to deploy since publish-profile is detected.
Setting SCM_DO_BUILD_DURING_DEPLOYMENT in Kudu container to true
Update using context.kuduService.updateAppSettingViaKudu
Response with status code 204
App setting SCM_DO_BUILD_DURING_DEPLOYMENT propagated to Kudu container
Setting ENABLE_ORYX_BUILD in Kudu container to true
Update using context.kuduService.updateAppSettingViaKudu
Response with status code 204
App setting ENABLE_ORYX_BUILD propagated to Kudu container
Package deployment using ZIP Deploy initiated.
Error: Failed to deploy web package to App Service.
Error: Execution Exception (state: PublishContent) (step: Invocation)
Error:   When request Azure resource at PublishContent, zipDeploy : Failed to use /home/runner/work/_temp/temp_web_package_0055.zip as ZipDeploy content
Error:     Failed to deploy web package to App Service.
Not Found (CODE: 404)
Error:       Error: Failed to deploy web package to App Service.
Not Found (CODE: 404)
    at Kudu.<anonymous> (/home/runner/work/_actions/Azure/functions-action/v1/lib/appservice-rest/Kudu/azure-app-kudu-service.js:23
    at Generator.throw (<anonymous>)
    at rejected (/home/runner/work/_actions/Azure/functions-action/v1/lib/appservice-rest/Kudu/azure-app-kudu-service.js:6:65)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
Error: Deployment Failed!
```

**Comment by martin-greentrax** [comment]
Could you solve this @kamranayub? I m seeing the same in our project since today. It was working for 2 years and also today and suddenly without any relevant changes it fails every deployment.

**Reply by kamranayub** [reply]
@martin-greentrax no unfortunately but mainly it's because the Az function I had won't work for my purposes anyway so I stopped deployments. Since then I have not looked into it further.

**Comment by martin-greentrax** [comment]
The same issue did go away for me after experimenting for a long time. Hope this helps someone else googling:

- I did add the environment variables that are suggested in other threads about the issue (like ENABLE_ORYX_BUILD and SCM_DO_BUILD_DURING_DEPLOYMENT) but can't say that they helped
- Go to your function on portal.azure.com and in the Overview blade at the bottom click on your Function "app"
I see this message there:
"Your app is currently in read only mode because you are running from a package file. To make any changes update the content in your zip file and WEBSITE_RUN_FROM_PACKAGE app setting."
- I clicked on Disable
- Then added WEBSITE_RUN_FROM_PACKAGE=0 in the blade Environment variables (it was not part of the variables before)
Restarted the function, enabled the "app" again and my github job went through again finally
